pkgs/leak_tracker_flutter_testing/lib/src/leak_tracking_for_tests.dart

```diff
@@ -0,0 +1,222 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:leak_tracker/leak_tracker.dart';
+import 'package:meta/meta.dart';
+
+void _emptyLeakHandler(Leaks leaks) {}
+
+/// Set of helper methods to adjust default leak tracker settings for tests.
+///
+/// Some metods modify [LeakTrackingForTests.settings], others return adjusted
+/// copy of [LeakTrackingForTests.settings].
+///
+/// Use methods, which modify [LeakTrackingForTests.settings], to
+/// change default settings for set of tests,
+/// for example for package or folder (in flutter_test_config.dart) or
+/// for library (in `setUpAll`).
+///
+/// Use methods that return adjusted [LeakTrackingForTests.settings]
+/// to customize default for an individual test:
+///
+/// ```dart
+/// testWidgets(
+///     'initialTimerDuration falls within limit',
+///     leakTracking: LeakTrackingForTests.withIgnoredLeaks(allNotGCed: true),
+///     (WidgetTester tester) async {
+///       ...
+/// ```
```

Comment:
The API and the way this class can be used feel super complex. Are all the different ways in which this class can be used necessary? What are the use cases that this enables?

Naively, I would have expected that we need to support two use cases:
1) Setting a default setting for all `testWidgets` that don't configure a local setting.
2) On a per test basis, `testsWidgets` should be able to override the defaults.

For that I would have expected something simple like: I can instantiate an immutable `LeakTrackingSettings` object. It's either configured by setting arguments in the `LeakTrackingSettings` constructor or I can invoke named constructors for some well known use cases (e.g. to track all leaks). I can then either pass that constructed object to `testWidgets` to configure something on a test-by-test basis, or I assign that object to a static (e.g. `LeakTrackingSettings.defaults`) to configure the defaults for all tests (e.g. from `flutter_test_config.dart`). 

The immutable `LeakTrackingSettings` object could have a copyWith method to create a setting derived from another. For example, if I just want to change one thing for a given test from the default settings I would do something like:

```dart
testWidgets('foo', leakTracking: LeakTrackingSettings.defaults.copyWith(barSetting: false), ...);
```

There could even be convenience copy methods for well-known use cases.

This would give `LeakTrackingSettings` a single, easy-to-understand API and it would make it explicit from what setting something is derived from.

Note: I am using `LeakTrackingSettings` as a placeholder name in the above. It could be called something else.